### DISCUSSION TITLE
Files incorrectly tagged as PureBasic

### DISCUSSION THREAD

**Discussion by bmaupin** [github-linguist/linguist#5628] [Classification] [answered]
I went to file a bug but it seems the preferred method is to start here.

I have a toy project which shows programming language trends. Upon noticing that `PureBasic` was trending (see https://bmaupin.github.io/langtrends/ > _Fastest growth_, sorry I haven't added routes for direct links yet), I checked to see what kind of projects were using PureBasic using this search:

https://github.com/search?q=language%3APureBasic&type=Repositories&ref=advsearch&l=PureBasic&l=

I browsed through a few of the results and the vast majority do not seem to be PureBasic. For example, many of the `.pb` hits are protobuf files:

https://github.com/HamedMP/tensorflow_export_cpp_example/search?l=purebasic
https://github.com/satyaborg/pose-estimation-detection/search?l=purebasic

These `.pbr` files, on the other hand, seem to be generated by IAR Embedded Workbench:

https://github.com/HauyuChen/Parking-System/search?l=purebasic

Of course this isn't an exhaustive list.

Thanks!

**Comment by lildude** [comment, marked as answer]
That'll be because the only language that Linguist knows about with the `.pb` extension is PureBasic so everything will get this language association by default 😁.

In order for them to be identified as another language, the extension would need to also be associated with another language. You can find details on adding support for an extension in the CONTRIBUTING.md file.

**Reply by bmaupin** [reply]
Ah, okay thanks for the explanation. I guess I thought that whatever heuristics were applied beyond just the file extension could be modified to filter out false positives like this. Thanks!
